catalog/nsili/catalog-nsili-common/src/main/java/org/codice/alliance/nsili/common/datamodel/NsiliDataModel.java

```diff
@@ -1342,24 +1349,25 @@ public EntityGraph getEntityGraph(String viewName) {
   private void updateMandatoryAttrs(String viewName, EntityNode[] viewNodes) {
     Map<String, List<String>> attrMap = new HashMap<>();
 
-    for (EntityNode entityNode : viewNodes) {
-      List<AttributeInformation> nodeAttrs = getAttributeInformation(entityNode.entity_name);
-      if (nodeAttrs != null) {
-        for (AttributeInformation nodeAttr : nodeAttrs) {
-          if (nodeAttr.mode == RequirementMode.MANDATORY) {
-            String attributeName = nodeAttr.attribute_name;
-            String[] attrNameArr = attributeName.split("\\.");
-            if (attrNameArr.length == 2) {
-              String parentNode = attrNameArr[0];
-              String attrName = attrNameArr[1];
-              List<String> attrs = attrMap.get(parentNode);
-              if (attrs == null) {
-                attrs = new ArrayList<>(4);
-                attrMap.put(parentNode, attrs);
-              }
-              attrs.add(attrName);
-            }
+    List<AttributeInformation> nodeAttrs =
+        Arrays.stream(viewNodes)
+            .map(e -> getAttributeInformation(e.entity_name))
```

Comment:
✏️ Couldn't this also be (just a nit, no need to change if you don't care to)
```java
List<AttributeInformation> nodeAttrs = Arrays.stream(viewNodes)
  .map(e -> e.entity_name) //is there a getter so we could do .map(e::getEntity_name)?
  .map(this::getAttributeInformation)
...
```

Reply:
will update, I think it looks cleaner to have a single action per line